design_documents/sncast_interface_overhaul.md

```diff
@@ -0,0 +1,47 @@
+# sncast interface overhaul
+
+## Context
+In light of needed changes on how the scripts and contract declaration work (ie adding support for scarb workspaces and various directory layouts),
+we need to re-think how we call the respective subcommands, and try to improve it. This design doc covers only a portion
+(although significant one) of sncast interface - it was decided recently we should make the interfaces as 'clap-native'
+as possible, but this design doc won't cover those changes in detail, as they are pretty straight forward and already have
+an opened issue.
+
+## The 'now'
+Currently, when declaring a contract we require user to pass the contract name using `--contract-name` flag. This flag only
+takes the contract's name, so for example it is not possible to specify the package this contract was defined in (only
+achievable when passing `--path-to-scarb-toml`). 
+
+When running a script we require a positional argument, that is the script package name. Similarly to declare, we could
+pass `--path-to-scarb-toml` if we wanted to pick the manifest we wanted to use.
+
+These approaches are a bit different to each other, and have a common flaw in the shape of having to specify a manifest 
+file one wants to work with, when the package needs to be specified. This may seem quite quirky, as we probably could 
+assume that we're always in some workspace/package.
+
+## Proposed changes
+We could streamline the way we interact with scarb-related subcommands (declare, script at the moment of writing this)
+to look like this:
+
+```bash
+$ # this is the same module
+$ sncast script package_namea::path::to::script_module
+$ sncast script to::script_module
+$ sncast script script_module
+
+$ # this is the same contract
+$ sncast declare package_namea::path::to::contract
+$ sncast to::contract
+$ sncast script script_module
+```
+
+In case there is only one script module/contract with specified name in workspace, only this name can be used to target it.
+Otherwise, if it is ambiguous in the context of workspace, we should show all matches with full paths, to let user decide
+which one to use.
+
+This change would mean that from now on we will have unified interface for declare/script, always assume we're in some workspace/package, 
+the `--path-to-scarb-toml` flag can be removed.
```

Comment:
I just don't think it will be required to have it. We can still support the env var, but I would be inclined to remove `--path-to-scarb-toml` as we'll require to always be in some workspace. Main reason to add this flag was that that was not always the case.

Reply:
Okay, didn't know we want to remove it from all commands